### PR TITLE
use passed options

### PR DESCRIPTION
**Pull request by artushin**
PR as per https://github.com/golang/groupcache/issues/49

### CODE REVIEW

**Comment by pierrre** [comment]
Can we validate this PR?
It fixes #49 & #60 .
Thank you.

File: http.go [hunk 1]
```diff
@@ -46,12 +46,10 @@ type HTTPPool struct {
 	// If nil, the client uses http.DefaultTransport.
 	Transport func(Context) http.RoundTripper
 
-	// base path including leading and trailing slash, e.g. "/_groupcache/"
-	basePath string
-
 	// this peer's base URL, e.g. "https://example.net:8000"
 	self string
 
+	options     *HTTPPoolOptions
```
Comment (adg) [inline http.go]: This needn't (shouldn't) be a pointer.

Comment (adg) [inline http.go]: Call it "opts".

File: http.go [hunk 2]
```diff
@@ -105,7 +103,7 @@ func NewHTTPPoolOpts(self string, o *HTTPPoolOptions) *HTTPPool {
 	}
 
 	p := &HTTPPool{
```
Comment (adg) [inline http.go]: after this, do 

if o != nil {
  p.opts = *o
}

**Comment by pierrre** [comment]
Fixed by #63 , we can close this PR.

**Pull request closed without merging**